packages/eslint-plugin-next/src/rules/no-html-link-for-pages.ts

```diff
@@ -134,6 +137,8 @@ export = defineRule({
           return
         }
 
+        pageUrls ||= getUrlFromPagesDirectories('/', foundPagesDirs)
```

Comment:
does this account for new files? not familiar with the method

Reply:
Yup, the only change here is to not evaluate the list of files for every visit, instead evaluate them per execution